src/PAK-Samples/MapReferenceToPaper/Program.cs

```diff
@@ -3,131 +3,399 @@
 //   Licensed under the MIT License.
 //-----------------------------------------------------------------------
 
+using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using System.Net.Http;
 using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
 using System.Web;
 
 namespace MapReferenceToPaper
 {
 
 
-    class Program
+    public class Program
     {
-        // Avoid creating new HttpClient for each request
-        static HttpClient client = new HttpClient();
+        public static Config configuration;
 
         static void Main(string[] args)
         {
-            if (args == null || args.Length < 6)
+            if (args == null || args.Length < 3)
             {
                 Console.WriteLine("MapReferenceToPaper");
                 Console.WriteLine("Attempts to map academic reference strings composed from rows/columns in a TSV file to a Microsoft Academic Graph (MAG) paper entity using Project Academic Knowledge Interpret method");
                 Console.WriteLine();
                 Console.WriteLine("Usage:");
-                Console.WriteLine("MapReferenceToPaper.exe <input.tsv> <output.tsv> <subscription_key> <attributes_to_map> <additional_attributes_to_return> <columns_from_input>");
+                Console.WriteLine("MapReferenceToPaper.exe <input.tsv> <output.tsv> <configuration.json>");
```

Comment:
>configuration [](start = 85, length = 13)

nit: could you use configuration manager to automatically load from an app.config file?

Reply:
Thanks for pointing this out Rick, but I'd like to keep the current model. If this is anything like the MAAdminTool's use of app settings re command line (having to first set the app setting you want to use as an environment variable) then I think it just adds too much complexity for who we're targeting with this.

Also, I'm not currently doing any custom parsing. I deserialize straight from the JSON file to my configuration class instance:
                    var configContent = File.ReadAllText(args[2]);
                    configuration = JsonConvert.DeserializeObject<Config>(configContent);

---
In reply to: [398900093](https://github.com/microsoft/mag-covid19-research-examples/pull/13#discussion_r398900093) [](ancestors = 398900093)

Reply:
We use an environment var in that tool to differentiate between two different files (dev / prod), you wouldn't need to do that here.  The suggestion was to use the .NET convention for config files, since this a .NET Framework app that would be adding config to the app.config file.  My example above was for .NET Core, didn't realize this was .NET Framework until later in the PR.

---
In reply to: [398934856](https://github.com/microsoft/mag-covid19-research-examples/pull/13#discussion_r398934856) [](ancestors = 398934856,398900093)